ccl/backupccl: TestFullClusterBackup failed

ccl/backupccl.TestFullClusterBackup [failed](https://tanzanite.cluster.engflow.com/invocations/default/53c68b1b-1577-4de5-aec0-4426ba0f4df2?testReportRun=9&testReportShard=36&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.2 @ [de49acf81407af360cea5a78cf66b808305fcf0d](https://github.com/cockroachdb/cockroach/commits/de49acf81407af360cea5a78cf66b808305fcf0d):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestFullClusterBackup (13m16s)
	TestFullClusterBackup/inSystemTenant=true (15s)

```
Stack: 

```
goroutine 568037 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestFullClusterBackup
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFullClusterBackup2715428574
    test_log_scope.go:76: use -show-logs to present logs inline
=== RUN   TestFullClusterBackup/inSystemTenant=false/zone_configs
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_all_databases_restored
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_that_tables_can_be_created_at_the_excepted_ID
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_user_table_data_restored
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_zones_are_restored_during_pre-restore
=== RUN   TestFullClusterBackup/inSystemTenant=false
    testutils.go:296: no Invalid Descriptors
    testutils.go:296: no Invalid Descriptors
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_all_schemas_are_restored
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_system_table_data_restored
=== RUN   TestFullClusterBackup/inSystemTenant=false/ensure_that_grants_are_restored
=== RUN   TestFullClusterBackup/inSystemTenant=true
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=36</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFullClusterBackup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45801

--- a/pkg/backup/full_cluster_backup_restore_test.go
+++ b/pkg/backup/full_cluster_backup_restore_test.go
@@ -142,11 +142,12 @@ CREATE TABLE data2.foo (a int);
 
 		// Setup the system systemTablesToVerify to ensure that they are copied to the new cluster.
 		// Populate system.users.
-		numBatches := 100
+		numBatches := 5
+		usersPerBatch := 20
 		if util.RaceEnabled {
 			numBatches = 1
+			usersPerBatch = 5
 		}
-		usersPerBatch := 10
 		userID := 0
 		for b := 0; b < numBatches; b++ {
 			sqlDB.RunWithRetriableTxn(t, func(txn *gosql.Tx) error {
@@ -193,7 +194,7 @@ CREATE TABLE data2.foo (a int);
 
 		// Create a bunch of user tables on the restoring cluster that we're going
 		// to delete.
-		numTables := 50
+		numTables := 10
 		if util.RaceEnabled {
 			numTables = 2
 		}

